docs/src/systems/implicit_incompressible_sph.md

```diff
@@ -0,0 +1,233 @@
+
+
+To derive the formulation for the linear system we start by discretizing the Continuity equation
+
+```math
+\frac{D\rho}{Dt} = - \rho \nabla \cdot v
+```
+
+For a particle $i$ a forward difference method is used to discretizie the left hand side of the equation to get
+
+```math
+\frac{\rho_i(t+ \Delta t - \rho_i(t))}{\Delta t}.
+```
+
+The right-hand side is discretized by the difference formulation of the divergence of the velocity field.
+
+```math
+\nabla \cdot \textbf{v}_i = \frac{1}{\rho_i} \sum_j m_j \textbf{v}_{ij} \nabla W_{ij},
+```
+where $\textbf{v}_{ij}$ is equal to $\textbf{v}_i - \textbf{v}_j$.
+
+So all in all the following discretized version of the continuity equation for a particle $i$ is achieved:
+
+```math
+\frac{\rho_i(t + \Delta t) - \rho_i(t)}{\Delta t} = \sum_j m_j \textbf{v}_{ij}(t+\Delta t) \nabla W_{ij}
+```
+
+In this implicit formulation the unknown velocities $\textbf{v}_{ij}(t + \Delta t)$ are needed.
+
+They are given by adding all pressure and non-pressure accelerations to the current velocity:
+
+```math
+\textbf{v}_i(t + \Delta t) = \textbf{v}_i(t) * \frac{\textbf{F}_i^{adv}(t) + \textbf{F}_i^p(t)}{m_i}
+```
+
+$\textbf{F}_i^{adv}$ are all non-pressure forces such as gravity, viscosiy, surface tension and more, while $\textbf{F}_i^p(t)$ are the unknown pressure forces, which have to be determined.
+
+Note that the IISPH is an incompressible fluid system, which means that the density of the fluid remain constant over time. By assuming a fixed reference density $\rho_0$ for all fluid particle over the whole time of the simulation, the density value at the next time step $\rho_i(t + \Delta t)$ also has to be this rest density. So $\rho_0$ can be plugged in for $\rho_i(t + \Delta t)$ in the formula above.
+
+The goal is to compute the pressure values required to obtain the pressure acceleration that ensures each particle reaches the rest density in the next time step. At the moment these pressure values are unknown in $t$, but all the non-pressure forces are already known.
+Therefore a predicted velocity can be calculated which depends only on the non-pressure forces $\textbf{F}_i^{adv}$:
+
+```math
+\textbf{v}_i^{adv}= \textbf{v}_i(t) + \Delta t \frac{\textbf{F}_i^{adv}(t)}{m_i},
+```
+Using this predicted velocity and the continuity equation a predicted density can also be determined.
+
+```math
+\rho_i^{adv} = \rho_i(t) + \Delta t \sum_j m_j \textbf{v}_{ij}^{adv} \nabla W_{ij}(t)
+```
+
+To achieve the rest density, the unknown pressure forces must counteract the compression caused by the non-pressure forces. In other words, they must resolve the deviation between the predicted density and the reference density.
+
+Therefore following equation needs to be fulfilled:
+```math
+\Delta t ^2 \sum_j m_j \left(  \frac{\textbf{F}\_i^p(t)}{m_i} - \frac{\textbf{F}\_j^p(t)}{m_j} \right) \nabla W_{ij}(t) = \rho_0 - \rho_i^{adv}
```

Comment:
Please render the docs and check if everything rendered correctly. This here doesn't:
<img width="583" height="100" alt="grafik" src="https://github.com/user-attachments/assets/90bbd9b4-595e-4bd4-a59d-eb4eed758356" />